### PR TITLE
Fix compilation errors with clang 19

### PR DESCRIPTION
**Pull request by marlamb**
The new compiler version seems to perform more exhaustive checks on templates, even if they are not initiated. As the code was malformed also before, it is likely that it is dead code. However, this commit is mainly to ensure users using clang 19 can include the header without compilation error.

Note that this commit does not guarantee that currently other code might not compile with clang 19, it just fixes some code which was used transitevily and caused problems.

### CODE REVIEW

**Comment by rcurtin** [comment]
Thank you, you are a hero!  I had seen these errors in the CRAN logs but had not yet been able to address them.  So this takes something off my todo list.  I may add the `Begin()` and `Count()` members to `Octree` for completeness-sake, but that can happen later.

I'll approve once I see the build succeeds (it should, you didn't make any breaking changes as far as I can tell.  Could take a few minutes though).

Thanks again! :rocket:

**Comment by marlamb** [comment]
Always an honor to contribute to a well maintained open source library!
On the CI it looks like a step building curl does not succeed. I guess this is unrelated to this commit?

**Review by rcurtin — APPROVED** [review]
Thanks again!  Feel free to add your name to the list of contributors and update the `HISTORY.md` file, or, I'll push a commit to `HISTORY.md` to document the change before merge. :+1:

**Comment by marlamb** [comment]
> Thanks again! Feel free to add your name to the list of contributors and update the `HISTORY.md` file, or, I'll push a commit to `HISTORY.md` to document the change before merge. 👍

Done, hope the very short description in the `HISTORY.md` is fine, as there still might be problems with `clang 19`, that I did not see transitively.

**Pull request merged**
